src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -1834,52 +1834,31 @@ private final void appendChars(CharSequence s, int off, int end) {
      *
      * @since 21
      * @throws IllegalArgumentException  if {@code count} is less than zero
-     * @throws StringIndexOutOfBoundsException  if the result overflows the buffer
+     * @throws IndexOutOfBoundsException  if the result overflows the buffer
      */
     public AbstractStringBuilder repeat(char c, int count) {
         if (count < 0) {
             throw new IllegalArgumentException("count is less than zero: " + count);
-        }
-        if (count == 0) {
+        } else if (count == 0) {
             return this;
         }
         ensureCapacityInternal(this.count + count);
-        if (isLatin1() && StringLatin1.canEncode(c)) {
-            int index = this.count;
-            while (count-- != 0) {
-                value[index++] = (byte) c;
+        int index = this.count;
+        int limit = index + count;
+        boolean isLatin1 = isLatin1();
+        if (isLatin1 && StringLatin1.canEncode(c)) {
+            while (index < limit) {
```

Comment:
```
while (index != limit) {
```
is probably safer, in particular if `limit` has overflown to `Integer.MIN_VALUE`.
Not sure if this could happen, but...